rust/codelist-rs/src/codelist.rs

```diff
@@ -519,6 +593,76 @@ mod tests {
 
         assert_eq!(codelist.metadata(), &metadata);
     }
+
+    #[test]
+    fn test_truncate_to_3_digits_snomed() -> Result<(), CodeListError> {
+        let metadata = create_test_metadata();
+
+        let mut snomed_codelist = CodeList::new(
+            "test_codelist".to_string(),
+            CodeListType::SNOMED,
+            metadata.clone(),
+            None,
+        );
+
+        // A SNOMED list is not truncatable
+        assert!(snomed_codelist.truncate_to_3_digits(TermManagement::First).is_err());
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_truncate_to_3_digits_icd10_4_digits() -> Result<(), CodeListError> {
+        let metadata = create_test_metadata();
+
+        let mut expected_codelist =
+            CodeList::new("test_codelist".to_string(), CodeListType::ICD10, metadata.clone(), None);
+        expected_codelist.add_entry(
+            "B01".to_string(),
+            "Varicella pneumonia".to_string(),
+            Some("B012 truncated to 3 digits".to_string()),
+        )?;
+
+        let mut observed_codelist =
+            CodeList::new("test_codelist".to_string(), CodeListType::ICD10, metadata.clone(), None);
+
+        observed_codelist.add_entry("B012".to_string(), "Varicella pneumonia".to_string(), None)?;
+
+        observed_codelist.truncate_to_3_digits(TermManagement::First)?;
+
+        assert_eq!(observed_codelist, expected_codelist);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_truncate_to_3_digits_3_and_4_digits() -> Result<(), CodeListError> {
+        let metadata = create_test_metadata();
+
+        let mut expected_codelist =
+            CodeList::new("test_codelist".to_string(), CodeListType::ICD10, metadata.clone(), None);
+        expected_codelist.add_entry(
+            "B01".to_string(),
+            "Varicella [chickenpox]".to_string(),
+            None,
+        )?;
+
+        let mut observed_codelist =
+            CodeList::new("test_codelist".to_string(), CodeListType::ICD10, metadata.clone(), None);
+
+        observed_codelist.add_entry(
+            "B01".to_string(),
+            "Varicella [chickenpox]".to_string(),
+            None,
+        )?;
+        observed_codelist.add_entry("B012".to_string(), "Varicella pneumonia".to_string(), None)?;
+
+        observed_codelist.truncate_to_3_digits(TermManagement::First)?;
+
+        assert_eq!(observed_codelist, expected_codelist);
+
+        Ok(())
```

Comment:
Good range of tests

Reply:
I like how cargo lets us run just these new tests with

    cargo test test_truncate_to_3_digits
    
I was running just `cargo test` at first and getting too much output.